### PR TITLE
le fishe update (carp gags and Cayenne can carry nuke disk)

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

Converts space carps to GAGS.
Cayenne can carry nuke disk and prime the nuke.

Ported:
https://github.com/tgstation/tgstation/pull/58710
https://github.com/tgstation/tgstation/pull/59614
https://github.com/tgstation/tgstation/pull/65268
https://github.com/tgstation/tgstation/pull/57395

## Testing Photographs and Procedure

I spawned in tested everything it works 

## Changelog
:cl: Arkatos1, JohnFulpWillard, Fikou
add: Carps now support GAGS.
add: Cayenne can carry nuke disk!
add: Achievement for picking up a nuke disk as Cayenne!
/:cl:

### CODE REVIEW

File: code/datums/achievements/misc_achievements.dm [hunk 1]
```diff
@@ -141,3 +141,9 @@
 	name = "Little Chickadee"
 	desc = "No, well, ten beers"
 	database_id = MEDAL_DRUNK
+
+/datum/award/achievement/misc/cayenne_disk
```
Comment (ike709) [inline code/datums/achievements/misc_achievements.dm:145]: No beecoin reward?

**Comment by ike709** [comment]
Why the Cayenne map removal?

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 2]
```diff
@@ -88,5 +145,88 @@
...
+		. += "<span class='notice'>Wait... is that [disky] in [p_their()] mouth?</span>"
+
+/mob/living/simple_animal/hostile/carp/cayenne/AttackingTarget()
+	if(istype(target, /obj/item/disk/nuclear))
+		var/obj/item/disk/nuclear/potential_disky = target
+		if(potential_disky.anchored)
+			return
+		potential_disky.forceMove(src)
+		disky = potential_disky
+		to_chat(src, "<span class='nicegreen'>YES!! You manage to pick up [disky]. (Click anywhere to place it back down.)</span>")
```
Comment (ike709) [inline code/modules/mob/living/simple_animal/hostile/carp.dm:183]: Can't they tell if it's fake? Seems like the message should change if it is.

File: code/datums/achievements/misc_achievements.dm (hunk 1 above)
Comment (Archanial) [reply code/datums/achievements/misc_achievements.dm:145]: The other achievements don't have a reward set too.

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 3]
```diff
@@ -88,5 +145,88 @@
...
+		var/obj/machinery/nuclearbomb/nuke = target
+		nuke.ui_interact(src)
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/carp/cayenne/Exited(atom/movable/AM, atom/newLoc)
+	. = ..()
+	if(AM == disky)
+		disky = null
+		update_icon()
```
Comment (ike709) [inline code/modules/mob/living/simple_animal/hostile/carp.dm:207]: How does this even happen?

File: code/modules/mob/living/simple_animal/hostile/carp.dm (hunk 2 above)
Comment (Archanial) [reply code/modules/mob/living/simple_animal/hostile/carp.dm:183]: Can the nuke ops tell?
Unless it's obviously fake, they can't.

File: code/modules/mob/living/simple_animal/hostile/carp.dm (hunk 3 above)
Comment (Archanial) [reply code/modules/mob/living/simple_animal/hostile/carp.dm:207]: force move

File: code/datums/achievements/misc_achievements.dm (hunk 1 above)
Comment (ike709) [reply code/datums/achievements/misc_achievements.dm:145]: Scroll up further.

Comment (Archanial) [reply code/datums/achievements/misc_achievements.dm:145]: 🙄

File: code/datums/achievements/misc_achievements.dm [hunk 4]
```diff
@@ -141,3 +141,9 @@
...
+	name = "Very Important Piscis"
+	desc = "You can rest well now."
+	database_id = MEDAL_CAYENNE_DISK
+	icon = "cayenne_disk"
```
Comment (Archanial) [inline code/datums/achievements/misc_achievements.dm:149]:
```suggestion
	icon = "cayenne_disk"
	reward = 2500
```

File: code/modules/mob/living/simple_animal/hostile/carp.dm (hunk 2 above)
Comment (ike709) [reply code/modules/mob/living/simple_animal/hostile/carp.dm:183]: I mean you gave them `TRAIT_DISK_VERIFIER`. Isn't that what the trait does?

**Review by ike709 — CHANGES_REQUESTED** [review]
LGTM besides the other comments

File: code/modules/mob/living/simple_animal/hostile/carp.dm (hunk 2 above)
Comment (Archanial) [reply code/modules/mob/living/simple_animal/hostile/carp.dm:183]: Probably it does, still, they can just check the disky itself, I don't see why I should make here another typecheck for fake

**Pull request merged**